R/commit.R

```diff
@@ -70,14 +74,25 @@ git_commit <- function(message, author = NULL, committer = NULL, repo = '.'){
 #' @rdname commit
 git_commit_all <- function(message, author = NULL, committer = NULL, repo = '.'){
   repo <- git_open(repo)
-  stat <- git_status(repo)
-  changes <- stat$file[!stat$staged & stat$status %in% c("modified", "renamed", "typechange")]
+  unstaged <- subset(git_status(repo), subset = !staged)
+
+  changes <- unstaged$file[unstaged$status %in% c("modified", "renamed", "typechange")]
   if(length(changes))
     git_add(changes, repo = repo)
-  deleted <- stat$file[!stat$staged & stat$status == "deleted"]
+
+  deleted <- unstaged$file[unstaged$status == "deleted"]
   if(length(deleted))
     git_rm(deleted, repo = repo)
-  git_commit(message = message, author = author, committer = committer, repo = repo)
+
+  new_files <- unstaged$file[unstaged$status == "new"]
+  if (length(new_files)) {
+    message("New, untracked files found, which won't be committed. Use git_add() instead.")
```

Comment:
This is new. It feels a little asymmetric to message about this but to not say anything about our other actions. But for a magic function like this, maybe messaging  is appropriate? Maybe the other branches should message.